### PR TITLE
Respect desired unit in variable with expression.

### PR DESCRIPTION
**Pull request by schymans**
If we define a new variable based on other variables, we want to be able to prescribe the desired units rather than taking the base units derived from the embedded expression. Example:
```
    class local_definition(Variable):
        """Local definition."""
        unit = joule

        class local_variable(Variable):
            """Local variable."""
            unit = kilogram * meter ** 2 / second ** 2
            default = 2

        expr = 1.5 * local_variable
```
With the current PR, `local_definition.definition.unit == joule` instead of `kilogram * meter ** 2 / second ** 2`

### CODE REVIEW

File: essm/variables/units.py [hunk 1]
```diff
@@ -23,20 +23,46 @@
...
+tesla = u.tesla
+volt = u.volt
 watt = u.watt
+weber = u.weber
 
 SI_DIMENSIONS = {
-    str(Quantity.get_dimensional_expr(d)): d for d in SI._base_units}
+    Quantity.get_dimensional_expr(d): d for d in SI._base_units}
+
+extunits = list(SI._base_units) + [
```
Comment (jirikuncar) [inline essm/variables/units.py]: can you rename it to something like: `SI_UNITS_EXT` or even better to replace `EXT` with full name.

Comment (schymans) [reply essm/variables/units.py]: Done, thanks!

**Comment by schymans** [comment]
I just found a problem with the new `derive_unit()`. Need to explore a bit more...

**Comment by schymans** [comment]
OK, solved the problem and added tests.

**Review by jirikuncar — DISMISSED** [review]

**Review by jirikuncar — APPROVED** [review]

**Pull request merged**